src/main/java/edu/ie3/datamodel/io/connectors/SqlConnector.java

```diff
@@ -115,21 +121,47 @@ public void shutdown() {
   }
 
   /**
-   * Extracts all field to value maps from the ResultSet, one for each row
+   * Method to execute a {@link PreparedStatement} and return its result as a stream.
    *
-   * @param rs the ResultSet to use
-   * @return a list of field maps
+   * @param ps to execute
+   * @param fetchSize used for {@link PreparedStatement#setFetchSize(int)}
+   * @return a stream of maps
+   * @throws SQLException if an exception occurred while executing the query
    */
-  public List<Map<String, String>> extractFieldMaps(ResultSet rs) {
-    List<Map<String, String>> fieldMaps = new ArrayList<>();
-    try {
-      while (rs.next()) {
-        fieldMaps.add(extractFieldMap(rs));
+  public Stream<Map<String, String>> toStream(PreparedStatement ps, int fetchSize)
+      throws SQLException {
+    ps.setFetchSize(fetchSize);
+    Iterator<Map<String, String>> sqlIterator = getSqlIterator(ps.executeQuery());
+
+    return StreamSupport.stream(
+        Spliterators.spliteratorUnknownSize(
+            sqlIterator, Spliterator.NONNULL | Spliterator.IMMUTABLE),
+        true);
+  }
+
+  /**
+   * Returns an {@link Iterator} for the given {@link ResultSet}.
+   *
+   * @param rs given result set
+   * @return an iterator
+   */
+  public Iterator<Map<String, String>> getSqlIterator(ResultSet rs) {
```

Comment:
Looking at the StackOverflow answer that I linked in the issue, isn't the purpose of this iterator to close the resources once we've iterated through the results?